src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -289,6 +342,9 @@ private void emitMetrics(long currWindowStartTime) throws Exception {
 
     metricsDB.commit();
     metricsDBMap.put(prevWindowStartTime, metricsDB);
+    if (batchMetricsEnabled) {
+      batchMetricsDBSet.add(prevWindowStartTime);
```

Comment:
SO we store the snapshot everytime we emit metrics, at query time we clone the snapshot (so we double the snapshot) and then we trim it. This is very memory intensive. We might just want to read the files directly from disk when the API is called ?

Reply:
We do not clone the snapshot at query time. We read the file itself. We use the batchMetricsDBSet to keep track of the timestamps of all the available dbs we have on disk, and then when we want to make use of these files during query handling, we call MetricsDB.fetchExisting, which simply creates a new connection to that on-disk database.

Reply:
This is not what we do. We add just the timestamp to a set. Then, the query handler uses the timestamps from this set to call fetchExisting. fetchExisting creates an additional connection to an already existing database (we do not clone the snapshot), and we access the data via that connection.